lib/node_modules/@stdlib/_tools/changelog/generate/lib/release_summary.js

```diff
@@ -0,0 +1,113 @@
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var groupBy = require( '@stdlib/utils/group-by' );
+var map = require( '@stdlib/utils/map' );
+var formatCommitFactory = require( './format_commit_factory.js' );
+var formatContributors = require( './format_contributors.js' );
+var formatCommits = require( './format_commits.js' );
+var breakingChanges = require( './breaking_changes.js' );
+var closedIssues = require( './closed_issues.js' );
+var sectionStart = require( './section_start.js' );
+var sectionEnd = require( './section_end.js' );
+var heading = require( './heading.js' );
+
+
+// MAIN //
+
+/**
+* Generates a release summary.
+*
+* @private
+* @param {ObjectArray} commits - commit objects
+* @param {boolean} excludeCommits - boolean indicating whether to exclude commit details
+* @param {boolean} excludeContributors - boolean indicating whether to exclude a list of contributors
+* @returns {string} summary
+*/
+function releaseSummary( commits, excludeCommits, excludeContributors ) {
+	var groupedFormat;
+	var grouped;
+	var out;
+
+	if ( !commits || !commits.length ) {
+		return '';
+	}
+	grouped = groupBy( commits, groupByType );
+	groupedFormat = formatCommitFactory( false, false );
+
+	out = '';
+	if ( grouped.feat ) {
+		out += sectionStart( 'features' );
+		out += heading( 'Features', 3 );
+		out += map( grouped.feat, groupedFormat ).join( '\n' );
+		out += sectionEnd( 'features' );
+	}
+	if ( grouped.fix ) {
+		out += sectionStart( 'bug-fixes' );
+		out += heading( 'Bug Fixes', 3 );
+		out += map( grouped.fix, groupedFormat ).join( '\n' );
+		out += sectionEnd( 'bug-fixes' );
+	}
+	if ( grouped.perf ) {
+		out += sectionStart( 'performance' );
+		out += heading( 'Performance', 3 );
+		out += map( grouped.perf, groupedFormat ).join( '\n' );
+		out += sectionEnd( 'performance' );
+	}
+	if ( grouped.revert ) {
+		out += sectionStart( 'reverts' );
+		out += heading( 'Reverts', 3 );
+		out += map( grouped.revert, groupedFormat ).join( '\n' );
+		out += sectionEnd( 'reverts' );
+	}
+	if ( grouped.deprecate ) {
+		out += sectionStart( 'deprecations' );
+		out += heading( 'Deprecations', 3 );
+		out += map( grouped.deprecate, groupedFormat ).join( '\n' );
+		out += sectionEnd( 'deprecations' );
+	}
+	out += breakingChanges( commits );
+	out += closedIssues( commits );
+	if ( !excludeCommits ) {
+		out += formatCommits( commits );
+	}
+	if ( !excludeContributors ) {
+		out += formatContributors( commits );
+	}
+	return out;
+
+	/**
+	* Groups commits by type.
+	*
+	* @private
+	* @param {Object} commit - commit object
+	* @returns {(string|null)} group key
+	*/
+	function groupByType( commit ) {
```

Comment:
This function does not depend on variables in the current scope. This should be moved to parent scope.

Reply:
Done.